tests/test_converter.py

```diff
@@ -0,0 +1,1220 @@
+            qml.Hadamard(2),
+            qml.Hadamard(4),
+            qml.PauliX(4),
+            qml.PauliY(1),
+            qml.RZ(0.34, wires=[1]),
+        ]
+
+        for converted, expected in zip(rec.queue, expected_queue):
+            assert converted.name == expected.name
+            assert converted.wires == expected.wires
+            assert converted.params == expected.params
+
+    def test_load_quil_via_entry_point(self):
+        """Test that a quil string can be loaded via the load entrypoint."""
+        quil_str = textwrap.dedent(
+            """
+            H 0
+            RZ(0.34) 1
+            CNOT 0 3
+            H 2
+            H 7
+            X 7
+            Y 1
+            RZ(0.34) 1
+        """
+        )
+
+        with OperationRecorder() as rec:
+            qml.load(quil_str, format="quil")()
+
+        # The wires should be assigned as
+        # 0  1  2  3  7
+        # 0  1  2  3  4
+
+        expected_queue = [
+            qml.Hadamard(0),
+            qml.RZ(0.34, wires=[1]),
+            qml.CNOT(wires=[0, 3]),
+            qml.Hadamard(2),
+            qml.Hadamard(4),
+            qml.PauliX(4),
+            qml.PauliY(1),
+            qml.RZ(0.34, wires=[1]),
+        ]
+
+        for converted, expected in zip(rec.queue, expected_queue):
+            assert converted.name == expected.name
+            assert converted.wires == expected.wires
+            assert converted.params == expected.params
+
+    def test_load_quil_file_via_entry_point(self):
+        """Test that a quil file can be loaded via the load entrypoint."""
+        cur_dir = os.path.dirname(os.path.abspath(__file__))
+
+        with OperationRecorder() as rec:
+            qml.load(os.path.join(cur_dir, "simple_program.quil"), format="quil_file")()
+
+        # The wires should be assigned as
+        # 0  1  2  3  7
+        # 0  1  2  3  4
+
+        expected_queue = [
+            qml.Hadamard(0),
+            qml.RZ(0.34, wires=[1]),
+            qml.CNOT(wires=[0, 3]),
+            qml.Hadamard(2),
+            qml.Hadamard(4),
+            qml.PauliX(4),
+            qml.PauliY(1),
+            qml.RZ(0.34, wires=[1]),
+        ]
+
+        for converted, expected in zip(rec.queue, expected_queue):
+            assert converted.name == expected.name
+            assert converted.wires == expected.wires
+            assert converted.params == expected.params
+
+    @pytest.mark.parametrize("angle", [0.0, 0.3, 0.5, 0.7, -0.2, 2.4])
+    def test_program_in_qnode(self, angle):
+        """Test how a simple program is used inside a QNode."""
+        program = pyquil.Program()
+
+        delta = program.declare("delta", "REAL")
+
+        program += g.RX(delta, 0)
+        program += g.CNOT(0, 1)
+
+        loader = qml.load(program, format="pyquil_program")
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(a):
+            loader({delta: a})
```

Comment:
Maybe it would be nice to add a test case in which an error arises since the variables cannot be mapped to the corresponding values (e.g. the types do not match).

Reply:
I did intentionally ignore the types of the variables. In pennylane we also only support floating point parameters, and in the quil code they can also only be used in places where only floating point parameters are accepted. So I think this will not be an issue.

Reply:
Sure, that makes sense! The idea that I was getting at would just be to include an integration test where an exception is raised

Reply:
Done!